File: practice/algorithms/implementation/save_the_prisoner/SaveThePrisoner.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    for(let i = 1; i < lines.length; i++) {
        console.log(savedPrisoner(lines[i]));
    }
};

const savedPrisoner = (line) => {
    const numbers = line.split(" ").map(i => parseInt(i));
    const prisoners = numbers[0];
    const sweets = numbers[1];
    const startId = numbers[2];

    const saved = (((sweets % prisoners) + startId) % prisoners) - 1;
    if (saved == 0) {
        return prisoners;
    } else {
        return saved;
    }
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/save_the_prisoner/SaveThePrisoner.java
import java.util.*;

public class SaveThePrisoner {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        for(int i = 0; i < tests; i++) {
            System.out.println(savedPrisoner(stdin.nextLine()));
        }
        stdin.close();
    }

    private static long savedPrisoner(String line) {
        String[] numbers = line.split(" ");
        long prisoners = Long.valueOf(numbers[0]);
        long sweets = Long.valueOf(numbers[1]);
        long startId = Long.valueOf(numbers[2]);

        long saved = (((sweets % prisoners) + startId) % prisoners) - 1;
        if (saved == 0) {
            return prisoners;
        } else {
            return saved;
        }
    }
}